src/index.js

```diff
@@ -8,5 +8,6 @@ export { default as getQueryKey } from './lib/get-query-key';
 export { default as queriesReducer } from './reducers/queries';
 export { default as entitiesReducer } from './reducers/entities';
 export { default as queryMiddleware } from './middleware/query';
+export { default as queryMiddlewareAdvanced } from './middleware/query-advanced';
```

Comment:
The problem I see with this is that projects that use the new `queryMiddlewareAdvanced` will keep pulling in `superagent` through the `queryMiddleware` from above.

I'm thinking we can easily fix this by:
- duplicating `src/index.js` to a new file `src/advanced.js`
- removing the `'./middleware/query'` import in `src/advanced.js`
- removing the `'./middleware/query-advanced'` import in `src/index.js` (to distinguish the different entries and usages)
- add a file to the root of the project (`advanced.js`) that is a CommonJS alias to `dist/commonjs/advanced`

This is sort of brittle because a single import for regular `redux-query` will cause the app's bundler to start including superagent, but I think it's acceptable for now. What do you think?